PATCHとDELETEのAPI作成

yoshiki2016 opened the pull request.

yoshiki2016 committed:
PATCHとDELETEのAPI作成

TomooAkamatsu commented on src/main/java/com/example/kadai7/PetController.java:
```diff
@@ -21,4 +19,22 @@ public ResponseEntity<String> createPet(@RequestBody @Valid PetCreateForm petCre
+    }
+
+    @DeleteMapping("/pet/delete/{id}")
+    public ResponseEntity<String> deletePet(@RequestBody @Valid PetCreateForm petCreateForm, @PathVariable(name = "id") int id) {
```
deleteの場合は、削除対象のリソースを一意に特定できるIDをパスパラメーターで渡すだけでいいので、リクエストボディは必要なさそうです！

yoshiki2016 replied:
リクエストボディの記述削除しました。

TomooAkamatsu replied:
修正内容LGTMです！

de992bd109d2626c0fca38fe280d1d9a521bfac2
のようにコミットハッシュを貼っていただくとやり取りがスムーズです👍

TomooAkamatsu commented on src/main/java/com/example/kadai7/PetController.java:
```diff
@@ -21,4 +19,22 @@ public ResponseEntity<String> createPet(@RequestBody @Valid PetCreateForm petCre
     }
+
+    @PatchMapping("/pet/edit/{id}")
+    public ResponseEntity<String> updatePet(@RequestBody @Valid PetCreateForm petCreateForm, @PathVariable(name = "id") int id) {
```
`PetCreateForm`なのでPatchも対応する場合は名前を変更するか、Patch用に新たにクラスを作成したほうがいいですね。

yoshiki2016 replied:
Patch用の新規クラスを作りました。

yoshiki2016 replied:
こちらで対応しました。
de992bd109d2626c0fca38fe280d1d9a521bfac2

TomooAkamatsu commented on src/main/java/com/example/kadai7/PetController.java:
```diff
@@ -21,4 +19,22 @@ public ResponseEntity<String> createPet(@RequestBody @Valid PetCreateForm petCre
+                .path("/pet/{id}") // id部分は実際に登録された際に発⾏したidを設定する
+                .build()
+                .toUri();
+        return ResponseEntity.created(url).body("PATCH通信成功");
```
patchの場合は、`ResponseEntity.created(url)`は適切ではなさそうですね。
`RESTAPI ステータス`といったキーワードでPatch処理が成功した場合のステータスを調べて実装してみてください👍

yoshiki2016 replied:
`ResponseEntity.created(url)`を廃止しました。以下で対応してます。
7806ed0439db3b3561ed92e3355f30bbd88f34c0

TomooAkamatsu commented on src/main/java/com/example/kadai7/PetController.java:
```diff
@@ -21,4 +19,22 @@ public ResponseEntity<String> createPet(@RequestBody @Valid PetCreateForm petCre
+                .path("/pet/{id}") // id部分は実際に登録された際に発⾏したidを設定する
+                .build()
+                .toUri();
+        return ResponseEntity.created(url).body("DELETE通信成功");
```
deleteの場合も同様にレスポンスの形式を修正してみてください！

yoshiki2016 replied:
`ResponseEntity.created(url)`を廃止しました。以下で対応してます。
7806ed0439db3b3561ed92e3355f30bbd88f34c0

TomooAkamatsu commented on src/main/java/com/example/kadai7/PetController.java:
```diff
@@ -21,4 +19,22 @@ public ResponseEntity<String> createPet(@RequestBody @Valid PetCreateForm petCre
         return ResponseEntity.created(url).body("POST通信成功");
     }
+
+    @PatchMapping("/pet/edit/{id}")
```
URIの中に`edit`や`delete`といった動詞が入るのはアンチパターンです。
`RESTAPI URI 設計`と調べてみてください👍

yoshiki2016 replied:
URIで`edit`,`delete`という単語を削除しました。
7806ed0439db3b3561ed92e3355f30bbd88f34c0

yoshiki2016 committed:
以下2つの指摘事項対応
https://github.com/yoshiki2016/raisetech_kadai7/pull/3#discussion_r1208356341
https://github.com/yoshiki2016/raisetech_kadai7/pull/3#discussion_r1208356528

yoshiki2016 committed:
3つの指摘事項に対応
https://github.com/yoshiki2016/raisetech_kadai7/pull/3#discussion_r1208358780
https://github.com/yoshiki2016/raisetech_kadai7/pull/3#discussion_r1208358889
https://github.com/yoshiki2016/raisetech_kadai7/pull/3#discussion_r1208365630
